### PR TITLE
Location file patch

### PR DESCRIPTION
**Pull request by widnyana**
Short Description: 

this patch fix issue for missing `location-username.json` and submodule init issue

Fixes:
- submodule init
- create location-%s.json if it doesnt exists

please review :)

### CODE REVIEW

**Comment by solderzzc** [comment]
This branch has conflicts that must be resolved

**Comment by SteffanLong** [comment]
Seems overly complex. We don't need to check if the file exists. Just write it. If it doesn't exist, it will after you create it. This is how the rest of the files work.

**Comment by widnyana** [comment]
@SteffanLong I've talked with `reaver01` in dev channel on slack, based on his suggestion i wrote this. :)

probably changing the file open mode from `w` to `w+` is the simplest way, any suggestion?

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "web"]
 	path = web
-	url = https://github.com/OpenPoGo/OpenPoGoWeb.git
```
Comment (solderzzc) [inline .gitmodules]: This line should not be touched.

Comment (widnyana) [inline .gitmodules]: @solderzzc noted, please leave any suggestion here. 

I'll create another pull request with more clean commits and close this one.

**Comment by solderzzc** [comment]
Sure, thanks a lot:)

**Pull request closed without merging**